### PR TITLE
PMM-7662: move to go modules

### PR DESCRIPTION
**Pull request by oter**
https://jira.percona.com/browse/PMM-7662

### CODE REVIEW

**Comment by BupycHuk** [comment]
Tests are failed

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,16 @@
+BASE_PATH = $(shell pwd)
+BIN_PATH := $(BASE_PATH)/bin
+
+export PATH := $(BIN_PATH):$(PATH)
+
+GOLANGCI_LINT := $(BIN_PATH)/golangci-lint
```
Comment (Dasio) [inline Makefile:6]: Is this necssary for each binary? We should agree on these, so we would be consistent across repositories

**Review by Dasio — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (oter) [reply Makefile:6]: it's discussable, but I see 2 pros in such way:
-  you don't need to specify `bin` folder
- a full path to binary is used

Comment (artemgavrilov) [reply Makefile:6]: I personally prefer to not use such small variables for every binary that we use. It leads to variable hell and increases Makefile complexity. IMHO it's not a big deal to copy-paste `bin/something` in several places, but have really simple, easy-to-read makefile.

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -28,7 +28,6 @@ jobs:
     env:
       GOPATH: ${{ github.workspace }}
       WORKING_DIR: ${{ github.workspace }}/src/github.com/percona/pmm-agent
-      GO111MODULE: off #TODO: remove once we move to go modules https://jira.percona.com/browse/PMM-7106
```
Comment (artemgavrilov) [inline .github/workflows/go.yml:31]: Let's add go modules cache to CI
https://github.com/percona-platform/platform/blob/77210c15d1db136a396d9f382b97c170877410d0/.github/workflows/ci.yml#L66-L72

and cache cleaner 
https://github.com/percona-platform/platform/blob/main/.github/workflows/clean.yml

**Review by artemgavrilov — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (BupycHuk) [reply Makefile:6]: Agree with @artemgavrilov

**Comment by BupycHuk** [comment]
@oter Tests are failing

File: Makefile (hunk 1 above)
Comment (oter) [reply Makefile:6]: as per discussion the variables removed

File: Makefile [hunk 3]
```diff
@@ -1,3 +1,8 @@
+BASE_PATH = $(shell pwd)
+BIN_PATH := $(BASE_PATH)/bin
+
+export PATH := $(BIN_PATH):$(PATH)
+
```
Comment (artemgavrilov) [inline Makefile:5]: Why do we need this?

File: .github/workflows/go.yml (hunk 2 above)
Comment (oter) [reply .github/workflows/go.yml:31]: added

**Comment by oter** [comment]
@BupycHuk tests are green now

File: Makefile (hunk 3 above)
Comment (oter) [reply Makefile:5]: are you about all the variables?

Comment (artemgavrilov) [reply Makefile:5]: Yep, we have several new variables, but nothing changed in fact.

Comment (artemgavrilov) [reply Makefile:5]: Discussed and resolved

File: .github/workflows/clean.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      - name: Enable Go build cache
+        uses: percona-platform/cache@v2
+        with:
+          path: ~/.cache/go-build
+          key: ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-${{ hashFiles('**') }}
+          restore-keys: |
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-${{ github.ref }}-
+            ${{ matrix.os }}-go-${{ matrix.go-version }}-build-
+      - name: Clean Go modules cache
+        run: go clean -modcache
```
Comment (artemgavrilov) [inline .github/workflows/clean.yml:72]: new line

**Review by artemgavrilov — APPROVED** [review]

File: .github/workflows/clean.yml (hunk 4 above)
Comment (oter) [reply .github/workflows/clean.yml]: added

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**